pkgs/development/python-modules/google-cloud-aiplatform/default.nix

```
@@ -0,0 +1,218 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+  gitUpdater,
+
+  # dependencies
+  google-api-core,
+  google-auth,
+  proto-plus,
+  protobuf,
+  packaging,
+  google-cloud-storage,
+  google-cloud-bigquery,
+  google-cloud-resource-manager,
+  shapely,
+
+  # optional dependencies
+  # NB: Commented-out dependencies are either currently missing from nixpkgs,
+  # or only used in an optional extra which is currently missing another dependency.
+
+  # absl-py,
+  # cloudpickle,
+  docker,
+  docstring-parser,
+  # explainable-ai-sdk,
+  fastapi,
+  google-cloud-bigquery-storage,
+  # google-cloud-trace,
+  # google-vizier,
+  httpx,
+  immutabledict,
+  # langchain,
+  # langchain-core,
+  # langchain-google-vertexai,
+  # lit-nlp,
+  mlflow,
+  nltk,
+  numpy,
+  # openinference-instrumentation-langchain,
+  # opentelemetry-exporter-gcp-trace,
+  # opentelemetry-sdk,
+  pandas,
+  pyarrow,
+  pydantic,
+  # pytest-xdist,
+  pyyaml,
+  ray,
+  requests,
+  # scikit-learn,
+  sentencepiece,
+  setuptools,
+  starlette,
+  tensorboard-plugin-profile,
+  tensorflow,
+  # torch,
+  tqdm,
+  urllib3,
+  uvicorn,
+  werkzeug,
+# xgboost,
+# xgboost_ray,
+}:
+
+buildPythonPackage rec {
+  pname = "google-cloud-aiplatform";
+  version = "1.73.0";
+
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "googleapis";
+    repo = "python-aiplatform";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-24NNMJ/m+VyEg/544GTCShFe1zGxel2fNkbFtwh0zfs=";
+  };
+
+  dependencies = [
+    google-api-core
+    google-auth
+    proto-plus
+    protobuf
+    packaging
+    google-cloud-storage
+    google-cloud-bigquery
+    google-cloud-resource-manager
+    shapely
+  ] ++ google-api-core.optional-dependencies.grpc;
+
+  optional-dependencies = rec {
+
+    profiler-deps = [
```

```suggestion
    cloud-profiler = [
```
Please correspond to the key in `extras_require`.
https://github.com/googleapis/python-aiplatform/blob/0160d4928e6e1a254a2e594216ae67abf83c7a47/setup.py#L257-L280

Same for other optional dependencies.